openedx/features/redhouse_features/registration/models.py

```diff
@@ -28,3 +28,25 @@ class AdditionalRegistrationFields(models.Model):
 
     organization_type = models.CharField(blank=False, max_length=32, verbose_name='Organization Type',
                                          choices=ORGANIZATIONS_TYPES)
+
+    @staticmethod
+    def validate_data(update, additional_fields, data, field_errors):
+        for key in additional_fields:
+            if key == 'phone' and len(update[key]) > 16:
+                field_errors['phone'] = {
+                    "developer_message": u"Phone length must be in between 0 and 16",
+                    "user_message": u"Phone length must be in between 0 and 16"
+                }
+            elif key == 'sch_org' and len(update[key]) > 150:
+                field_errors['sch_org'] = {
+                    "developer_message": u"School / Organization length must be in between 0 and 150",
+                    "user_message": u"School / Organization length must be in between 0 and 150"
+                }
+            elif (key == 'organization_type' and update['organization_type'] not in zip(*ORGANIZATIONS_TYPES)[0] and
+                  update['organization_type'] is not None):
+                field_errors['organization_type'] = {
+                    "developer_message": u"Invalid organization type",
+                    "user_message": u"Invalid organization type"
+                }
+            else:
+                data[key] = update[key]
```

Comment:
We have to use if-else in loop. Can we instead a dict and place validator functions in that dict against field keys?